File: csharp/504.cs
public class Solution {
    public string ConvertToBase7(int num) {
        StringBuilder sb = new StringBuilder();
        bool isNeg = num < 0;
        num = num < 0 ? -num : num;

        while(num > 0){
            sb.Append(num % 7);
            num /= 7;
        }
        if(isNeg) { sb.Append("-"); }

        return sb.Length == 0 ? "0" : string.Join("", sb.ToString().Reverse());
    }
}

File: python/504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        sign = 1 if num < 0 else 0
        ans, num = "", abs(num)
        while(num):
            num, digit = divmod(num, 7)
            ans += str(digit)
        
        return "-"*sign + ans[::-1] if ans else "0"

